java/libphonenumber/test/com/google/i18n/phonenumbers/PhoneNumberUtilTest.java

```diff
@@ -2652,6 +2660,14 @@ public void testParseExtensions() throws Exception {
     assertEquals(usWithExtension, phoneUtil.parse("+1 (645) 123 1234-910#", RegionCode.US));
     // Retry with the same number in a slightly different format.
     assertEquals(usWithExtension, phoneUtil.parse("+1 (645) 123 1234 ext. 910#", RegionCode.US));
+
+    // Comma and semicolon are allowed in extensions
+    usWithExtension.clear();
+    usWithExtension.setCountryCode(1).setNationalNumber(6451231234L).setExtension("91,0");
+    assertEquals(usWithExtension, phoneUtil.parse("+1 (645) 123 1234 ext: 91,0", RegionCode.US));
+    usWithExtension.clear();
+    usWithExtension.setCountryCode(1).setNationalNumber(6451231234L).setExtension("91;0");
```

Comment:
Please exercise the allowed pattern a bit more.
(both allowed and disallowed)

- test which shows what happens where there's a trailing comma or semicolon.
- multiple control characters in a row (but not at the end)
- extension starting with a control character
- "92-3" and "91.0" i.e. what happens if it's another punctuation character appears in the extension